util/git/client.go

```diff
@@ -631,13 +631,17 @@ func (m *nativeGitClient) lsRemote(revision string) (string, error) {
 		revision = "HEAD"
 	}
 
-	semverSha, err := m.resolveSemverRevision(revision, refs)
-	if err != nil {
-		return "", err
-	}
-
-	if semverSha != "" {
-		return semverSha, nil
+	// Check if the revision is a valid semver constraint before attempting to resolve it
+	if constraint, err := semver.NewConstraint(revision); err == nil {
+		semverSha, err := m.resolveSemverRevision(constraint, refs)
+		if err != nil {
+			return "", err
+		}
```

Comment:
With the current change in the PR, the function `resolveSemverRevision` never returns an error.  Was this intended?